shared-libs/transitions/test/unit/transitions.js

```diff
@@ -333,5 +336,186 @@ describe('transitions', () => {
+      });
+
+      expect(result).to.equal(true);
+      expect(transition.filter.args).to.deep.equal([[change.doc, change.info]]);
+      expect(transition.onMatch.args).to.deep.equal([[change]]);
+      expect(infodoc.updateTransition.args).to.deep.equal([[change, transition.key, true]]);
+    });
+
+    it('should apply transition without change', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub().returns(true),
+        onMatch: sinon.stub().resolves(false),
+      };
+      sinon.stub(infodoc, 'updateTransition');
+      const change = { doc: {}, info: {} };
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+
+      expect(result).to.equal(false);
+      expect(transition.filter.args).to.deep.equal([[change.doc, change.info]]);
+      expect(transition.onMatch.args).to.deep.equal([[change]]);
+      expect(infodoc.updateTransition.called).to.equal(false);
+    });
+
+    it('should skip transitions that dont pass filter', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub().returns(false),
+        onMatch: sinon.stub(),
+      };
+      sinon.stub(infodoc, 'updateTransition');
+      const change = { doc: {}, info: {} };
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+
+      expect(result).to.equal(undefined);
+      expect(transition.filter.args).to.deep.equal([[change.doc, change.info]]);
+      expect(transition.onMatch.called).to.equal(false);
+      expect(infodoc.updateTransition.called).to.equal(false);
+    });
+
+    it('should skip transitions that cant run', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub(),
+        onMatch: sinon.stub(),
+      };
+
+      sinon.stub(infodoc, 'updateTransition');
+      const change = { deleted: true, doc: {}, info: {} };
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+
+      expect(result).to.equal(undefined);
+      expect(transition.filter.called).to.equal(false);
+      expect(transition.onMatch.called).to.equal(false);
+      expect(infodoc.updateTransition.called).to.equal(false);
+    });
+
+    it('should force running a transition', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub(),
+        onMatch: sinon.stub().resolves(true),
+      };
+
+      sinon.stub(transitions, 'canRun').returns(false);
```

Comment:
```suggestion
      sinon.stub(transitions, 'canRun');
```

Unless I am missing something, we don't need to set any behavior here since this should never be called.